server/src-lib/Hasura/RQL/DDL/Permission.hs

```diff
@@ -84,12 +85,12 @@ type InsPermDef = PermDef InsPerm
 type CreateInsPerm = CreatePerm InsPerm
 
 buildViewName :: QualifiedTable -> RoleName -> PermType -> QualifiedTable
-buildViewName (QualifiedObject sn tn) rn pt =
-  QualifiedObject hdbViewsSchema $ TableName
-  (roleNameToTxt rn <> "__" <> T.pack (show pt) <> "__" <> snTxt <> "__" <> tnTxt)
+buildViewName qt rn pt = QualifiedObject hdbViewsSchema tableName
   where
-    snTxt = getSchemaTxt sn
-    tnTxt = getTableTxt tn
+    tableName = TableName $ T.pack $ show hash
+    -- Blake2b_224 generates 56 character length hash
+    hash :: CH.Digest CH.Blake2b_224 =
```

Comment:
A thought: it isn’t immediately clear to someone reading this code *why* we’re doing this hashing. This might be a good situation for [something like GHC’s “notes” system](https://www.stackbuilders.com/news/the-notes-of-ghc). The idea is to create a long-form comment somewhere, like this:

```haskell
{- Note [Postgres identifier length limitations]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Postgres truncates identifiers to a maximum of 63 characters by default (see
https://www.postgresql.org/docs/12/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS). This causes
problems for A, B, and C because of X and Y, so we work around it by doing Z. -}
```

Except obviously `A`, `B`, `C`, etc. are replaced with real explanations. :) Then somewhere else in the code you can write

```haskell
-- see Note [Postgres identifier length limitations]
hash :: CH.Digest CH.Blake2b_224 =
```

wherever the note is relevant. That way, explanations for tricky concepts can be centralized in one place, and they can be referenced from other places in the code. I think there is already at least one or two places in `graphql-engine` where we do some work to work around Postgres’s identifier length limitation, so it would be great to put a note somewhere and add comments in all those places referencing it!